app/src/marketplace/components/ProductPage/Terms/index.jsx

```diff
@@ -0,0 +1,114 @@
+// @flow
+
+import React from 'react'
+import styled from 'styled-components'
+import { Translate, I18n } from 'react-redux-i18n'
+
+import ProductContainer from '$shared/components/Container/Product'
+import type { Product } from '$mp/flowtype/product-types'
+import { MD } from '$shared/utils/styled'
+
+const Container = styled(ProductContainer)`
+    background-color: #f8f8f8;
+    margin-top: 48px !important;
+    
+    @media (max-width: ${MD}px) {
+        padding-left: 0 !important;
+        padding-right: 0 !important;
+    }
+`
+
+const Header = styled.div`
+    display: flex;
+    background-color: #efefef;
+    color: #525252;
+    font-size: 14px;
+    font-weight: 500;
+    line-height: 16px;
+    letter-spacing: 1.17px;
+    text-transform: uppercase;
+    height: 4.5rem;
+    align-items: center;
+    padding: 2em;
+`
+
+const Content = styled.div`
+    padding: 1.5em 2em;
+`
+
+const Text = styled.span`
+    font-size: 14px;
+    line-height: 28px;
+    font-weight: ${(props) => (props.bold ? '500' : '400')};
+`
+
+type Props = {
+    className?: string,
+    product: Product,
+}
+
+const getTermStrings = (ids: Array<string>) => (
+    ids.map((id, index) => {
+        const term = I18n.t(`editProductPage.terms.${id}`)
+        if (index !== 0) {
+            const separator = (index === ids.length - 1) ? ' & ' : ', '
+            return `${separator}${term.toLowerCase()}`
+        }
+        return term
+    })
+)
+
+const Terms = ({ className, product }: Props) => {
+    if (product == null) {
+        return null
+    }
+
+    const terms = product.termsOfUse
+    const entries = Object.entries(terms)
+    const permitted = entries.filter((e) => e[1] === true).map((e) => e[0])
```

Comment:
same as `Object.keys(terms).filter(Boolean)`?

Reply:
maybe you can memoize it here:
```js
const permitted = useMemo(() => getTermStrings(Object.keys(terms).filter(Boolean)), [terms])
```

Reply:
Memoization is a good idea. `Object.keys` does not do what we need here since we need to extract keys of values that are `true` (for permitted) and `false` (for not permitted).